code/controllers/subsystem/overlays.dm

```diff
@@ -47,8 +47,13 @@ SUBSYSTEM_DEF(overlays)
 	for (var/thing in queue)
 		count++
 		if(thing)
-			STAT_START_STOPWATCH
 			var/atom/A = thing
+			if(A.overlays.len >= MAX_ATOM_OVERLAYS)
+				//Break it real GOOD
+				A.overlays.Cut()
+				stack_trace("Too many overlays on [A.type] - [A.overlays.len], refusing to update and cutting")
```

Comment:
You did A.overlays.Cut() on the line above, A.overlays.len should be 0 by now.

Reply:
FUDGe